README.md

```diff
@@ -10,39 +10,146 @@
+
+#### Fixed values in configuration
+```
+token:
+  ...
+
+  claims:
+    capabilities
+      value:
+        - capability0
+        - capability1
+
+```
+The above config would create the claim: 
+``` 
+"capabilities":  ["capability0", "capability1"]
+```
+#### HTTP Header or Parameter 
+```
+token:  
+  ...
+
+  claims:
+    mac:
+      header: X-Midt-Mac-Address
+      parameter: mac
+```
+The value of the `mac` claim would come from the specified header or parameter name of the request to the `/issue` endpoint.
+
+#### PartnerID
+Although it is configured separately, it behaves very similarly to the previous source type.
+
+```
+partnerID:
+  claim: partner-id
+  metadata: pid # only needed when a remote claims server needs this value
+  header: X-Midt-Partner-ID
+  parameter: pid
+  default: comcast
+```
+
+#### Remote claims
+
+```
+remote:
+  method: "POST"
+  url: "http://remote-claims-server.example.com/claims"
+```
+For more informatiom on how to configure Themis to run as your remote claims server, read the next section on Remote Server Claims Configuration.
+
+
+### Remote Server Claims Configuration
+### Using Themis as the remote claims server
+You can do this by configuring only the `claims` server in your configuration file. 
+Claims are configured exactly the same as explained above.
+
+#### Sending data to remote server
+Suppose the remote claims server needs the ID of the device requesting a token in the form of an HTTP Header named `X-Midt-Device-Id`. The `token.metadata` configuration element allows you to specify which values are sent to the remote claims server.
+
+```
+token:
+  ...
+  metadata:
+    ID:
+      header: X-Midt-Device-Id
+```
+
+## Build
+There is a single binary for themis and its execution is fully driven by configuration.
+
+### Makefile
+
+The Makefile has the following options you may find helpful:
+* `make build`: builds the Themis binary
+* `make docker`: builds a docker image for themis
+* `make local-docker`: builds a docker image for themis with the `local` version tag
+* `make test`: runs unit tests with coverage for Themis 
+* `make clean`: deletes previously-built binaries and object files
+
+## Deploy
+At the simplest form, run the binary with the flag specifying the configuration file
+```
+./themis -f themis.yaml
+``` 
+
+### Docker
+We recommend using docker for local development.
+
+```
+# Build docker image for themis
+# themis.yaml specifies the static claims which will be returned in the JWT
+make local-docker
+
+# Run container service
+docker run -p 6501:6501 themis:local
+
+# Request a JWT token
+curl http://localhost:6701/issue -H 'X-Midt-Mac-Address: mac:1122334455'
 ```
-This `build.sh` script will build the binary and docker image
 
-## Usage
-Once everything is up and running you can start sending requests. Below are a few examples.
-TODO: Add examples
+Explore JWT contents at [jwt.io](https://jwt.io/)
```

Comment:
```suggestion
```
I think we should move this towards the top so someone who needs more context will see it sooner.